x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -246,12 +241,19 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 		}
 
 		// update provider payment storage with complainer's CU
-		err = k.updateProviderPaymentStorageWithComplainerCU(ctx, relay.UnresponsiveProviders, logger, epochStart, relay.SpecId, relay.CuSum, servicersToPair, projectID)
+		err = k.updateProviderPaymentStorageWithComplainerCU(ctx, relay.UnresponsiveProviders, logger, epochStart, relay.SpecId, relay.CuSum, servicersToPair, project.Index)
 		if err != nil {
 			utils.LogLavaEvent(ctx, logger, types.UnresponsiveProviderUnstakeFailedEventName, map[string]string{"err:": err.Error()}, "Error Unresponsive Providers could not unstake")
 		}
 	}
 
+	// event for rejected CU
+	var rejected_relays bool
```

Comment:
if all relays fail, fail the tx and return error

Reply:
done. see https://github.com/lavanet/lava/commit/f1535db03c1f18bbf62b5f42b518426f41d7fbef